src/value.rs

```diff
@@ -20,13 +20,7 @@ use crate::de::{Error as RonError, Result};
 pub struct Number(f64);
 
 impl Number {
-    /// Panics if `v` is not a real number
-    /// (infinity, NaN, ..).
     pub fn new(v: f64) -> Self {
-        if !v.is_finite() {
-            panic!("Tried to create Number with a NaN / infinity");
-        }
```

Comment:
I don't think we can just remove this; we can only implement `Ord` because of this check. Can you please check what behavior this causes when sorting `Number`?

Reply:
We don't implement Ord, only PartialOrd:
```
#[derive(Copy, Clone, Debug, PartialOrd, PartialEq)]
pub struct Number(f64);
```
and it should be fine as:
```
impl PartialOrd<f64> for f64
```

Reply:
Yes, you're right, it will fail in case of NaN:
```
impl Ord for Number {
    fn cmp(&self, other: &Self) -> Ordering {
        self.partial_cmp(other).expect("Bug: Contract violation")
    }
}
```
Probably we should not provide Ord implementation for Number.

Reply:
I don't think we can not provide the `Ord` implementation, because `Number` is a variant of `Value`, which in turn needs to implement `Ord` so we can use `Value` as key.

So, what we should do is just handle the `NaN` case and specify the ordering in the docs. For example, we might just always return `Less` for `NaN`.

Reply:
It means we also need to change Eq implementation and make two NaNs equals to each other. And also make Ord return Less if the right is Nan but the left is not, Equal is both are NaNs and Greater if the right is not NaN, but the left is. Correct?

Reply:
Implemented